packages/@aws-cdk/aws-apprunner-alpha/lib/auto-scaling-configuration.ts

```diff
@@ -0,0 +1,198 @@
+
+  /**
+   * The maximum number of instances that a service scales up to.
+   * At most maxSize instances actively serve traffic for your service.
+   *
+   * Must be between 1 and 25.
+   *
+   * @default 25
+   */
+  readonly maxSize?: number;
+
+  /**
+   * The minimum number of instances that App Runner provisions for a service.
+   * The service always has at least minSize provisioned instances.
+   *
+   *
+   * Must be between 1 and 25.
+   *
+   * @default 1
+   */
+  readonly minSize?: number;
+}
+
+/**
+ * Attributes for the App Runner Auto Scaling Configuration.
+ */
+export interface AutoScalingConfigurationAttributes {
+  /**
+   * The name of the Auto Scaling Configuration.
+   */
+  readonly autoScalingConfigurationName: string;
+
+  /**
+   * The ARN of the Auto Scaling Configuration.
+   */
+  readonly autoScalingConfigurationArn: string;
+
+  /**
+   * The revision of the Auto Scaling Configuration.
+   */
+  readonly autoScalingConfigurationRevision: number;
+}
+
+/**
+ * Represents the App Runner Auto Scaling Configuration.
+ */
+export interface IAutoScalingConfiguration extends cdk.IResource {
+  /**
+   * The ARN of the Auto Scaling Configuration.
+   * @attribute
+   */
+  readonly autoScalingConfigurationArn: string;
+
+  /**
+   * The Name of the Auto Scaling Configuration.
+   * @attribute
+   */
+  readonly autoScalingConfigurationName: string;
+
+  /**
+   * The revision of the Auto Scaling Configuration.
+   * @attribute
+   */
+  readonly autoScalingConfigurationRevision: number;
+}
+
+/**
+ * The App Runner Auto Scaling Configuration.
+ *
+ * @resource AWS::AppRunner::AutoScalingConfiguration
+ */
+export class AutoScalingConfiguration extends cdk.Resource implements IAutoScalingConfiguration {
+  /**
+   * Imports an App Runner Auto Scaling Configuration from attributes
+   */
+  public static fromAutoScalingConfigurationAttributes(scope: Construct, id: string,
+    attrs: AutoScalingConfigurationAttributes): IAutoScalingConfiguration {
+    const autoScalingConfigurationArn = attrs.autoScalingConfigurationArn;
+    const autoScalingConfigurationName = attrs.autoScalingConfigurationName;
+    const autoScalingConfigurationRevision = attrs.autoScalingConfigurationRevision;
+
+    class Import extends cdk.Resource {
+      public readonly autoScalingConfigurationArn = autoScalingConfigurationArn
+      public readonly autoScalingConfigurationName = autoScalingConfigurationName
+      public readonly autoScalingConfigurationRevision = autoScalingConfigurationRevision
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * Imports an App Runner Auto Scaling Configuration from its ARN
+   */
+  public static fromArn(scope: Construct, id: string, autoScalingConfigurationArn: string): IAutoScalingConfiguration {
+    const arn = cdk.Stack.of(scope).splitArn(autoScalingConfigurationArn, cdk.ArnFormat.SLASH_RESOURCE_NAME);
+
+    const resourceParts = arn.resourceName?.split('/');
```

Comment:
I think `arn.resourceName?.split('/');` won't work when `autoScalingConfigurationArn` is a Token.

[splitArn](https://github.com/aws/aws-cdk/blob/c1e1b4d777a65ba3fcb6a108c43b10b4e7b9ace6/packages/aws-cdk-lib/core/lib/stack.ts#L804) can handle token but split only handles string. I guess we need Fn.Split and Fn.Select.

Reply:
Thanks. I've fixed it.